ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh

```diff
@@ -30,14 +29,15 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
       server $name $ip:6443 check inter 1s
       server $name-v6 [$ipv6]:6443 check inter 1s"
   fi
-  if [ "$num_workers" -eq 0 ] || [[ "$name" =~ worker* ]]; then
-    INGRESS80="$INGRESS80
-      server $name $ip:80 check inter 1s
-      server $name-v6 [$ipv6]:80 check inter 1s"
-    INGRESS443="$INGRESS443
-      server $name $ip:443 check inter 1s
-      server $name-v6 [$ipv6]:443 check inter 1s"
-  fi
+  # add all hosts to ingress to workaround OCPQE-21041
+  # since ingress router pods rely on worker role instead of host name
+  # no risk to add all host to ingress since no other service listen on port 80/443
```

Comment:
@lihongan, if I'm not wrong, this is happening when workers=1, right? As in workers=1 implies the masters have the worker role too?

Reply:
If yes, I'd change this to have the minimum required change and a proper comment:

```
# Master nodes get the worker role when the cluster is installed with less than 2 workers.
if [ "$num_workers" -lt 2 ] ...
```

Reply:
The ratio is that we document the configuration of this load balancer for baremetal UPI. I know it should be harmless but still I would prefer keeping this configuration as closest as possible to what we state for users.

Reply:
Yes, ff the condition of `master nodes get the worker role when less than 2 workers (host name)` is always true then we can use this one
``` 
if [ "$num_workers" -lt 2 ] ...
```